jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -559,11 +560,16 @@ class CodeWhispererService {
     }
 
     fun canDoInvocation(editor: Editor, type: CodewhispererTriggerType): Boolean {
-        if (type == CodewhispererTriggerType.AutoTrigger && !CodeWhispererExplorerActionManager.getInstance().isAutoEnabled()) {
+        if (type == CodewhispererTriggerType.AutoTrigger && !CodeWhispererExplorerActionManager.getInstance().getCodeWhispererExplorerState(
+                CodeWhispererExploreStateType.IsAutoEnabled
```

Comment:
taking this as an example, I am not sure what benefit we are getting from using the `getCodeWhispererExplorerState` as the "entry point" and pass different parameter, to me this requires more knowledge for the caller, . Here we changed logic to " get the `isAutoEnabled` state". And now it's fine since `isAutoEnabled` state is equivalent to "whether cwspr is enabled-automatically" but when we want to change it's implementation in the future we have to change this everywhere instead of just in `isAutoEnabled()`. To me this is a lack of extendibility. So again I am wonder what's the purpose of this factor and what does it achieve.

Reply:
umm no specific purpose/context of the refactor, just thought if it will be cleaner but if it's adding the churm we can close this.

Reply:
I would hold this refactor for now for the extendibility reason above.